update/v1alpha1/types_update_status.go

```diff
@@ -0,0 +1,750 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// UpdateStatus is the API about in-progress updates, kept populated by Update Status Controller by
+// aggregating and summarizing UpdateInsights produced by update informers
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=updatestatuses,scope=Namespaced
+// +openshift:api-approved.openshift.io=TODO
+// +openshift:file-pattern=cvoRunLevel=0000_00,operatorName=cluster-version-operator,operatorOrdering=02
+// +openshift:enable:FeatureGate=UpgradeStatus
+// +kubebuilder:metadata:annotations="description=Provides health and status information about OpenShift cluster updates."
+// +kubebuilder:metadata:annotations="displayName=UpdateStatuses"
+type UpdateStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec UpdateStatusSpec `json:"spec"`
+	// +optional
+	Status UpdateStatusStatus `json:"status,omitempty"`
+}
+
+// UpdateStatusSpec is empty for now, can possibly hold configuration for Update Status Controller in the future
+type UpdateStatusSpec struct {
+}
+
+// +k8s:deepcopy-gen=true
+
+// UpdateStatusStatus is the API about in-progress updates, kept populated by Update Status Controller by
+// aggregating and summarizing UpdateInsights produced by update informers
+type UpdateStatusStatus struct {
+	// controlPlane contains a summary and insights related to the control plane update
+	ControlPlane ControlPlaneUpdateStatus `json:"controlPlane"`
+
+	// workerPools contains summaries and insights related to the worker pools update
+	WorkerPools []PoolUpdateStatus `json:"workerPools"`
```

Comment:
I'm not API-review enough to know why `Conditions` is a slice and not a map, or when it's suggested to use a `listType=map` slice instead of an actual map.  But as an API consumer trying to explain why pool A was having trouble updating, having `WorkerPools map[string]PoolUpdateStatus` seems like it would be more convenient.  But iterating over a slice and hunting for a unique name is obviously workable, if the API reviewers prefer that approach.

Reply:
This seems to be an established Kube convention (I agree with you about the UX).

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps

This place was actually missing the `listType=map` annotations! Thanks, added.

Reply:
Thanks for the doc link; you can mark this thread resolved :)